cmd/src/debug_common.go

```diff
@@ -0,0 +1,122 @@
+	f := &archiveFile{name: path}
+	f.data, f.err = exec.CommandContext(ctx, cmd, args...).CombinedOutput()
+	if f.err != nil {
+		f.err = errors.Wrapf(f.err, "executing command: %s %s: received error: %s", cmd, strings.Join(args, " "), f.data)
+	}
+	return f
+}
+
+// This function prompts the user to confirm they want to run the command
+func verify(confirmationText string) (bool, error) {
+	input := ""
+	for strings.ToLower(input) != "y" && strings.ToLower(input) != "n" {
+		fmt.Printf("%s [y/N]: ", confirmationText)
+		if _, err := fmt.Scanln(&input); err != nil {
+			return false, err
+		}
+	}
+
+	return strings.ToLower(input) == "y", nil
+}
+
+func processBaseDir(base string) (string, string) {
+	var baseDir string
+	if !strings.HasSuffix(base, ".zip") {
+		baseDir = base
+		base = base + ".zip"
+	} else {
+		baseDir = strings.TrimSuffix(base, ".zip")
+	}
+	return base, baseDir
+}
+
+// write all the outputs from an archive command passed on the channel to to the zip writer
+func writeChannelContentsToZip(zw *zip.Writer, ch <-chan *archiveFile, verbose bool) error {
+	for f := range ch {
+		if verbose {
+			log.Printf("archiving file %q with %d bytes", f.name, len(f.data))
+		}
+
+		zf, err := zw.Create(f.name)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create %s: %w", f.name, err)
+		}
+
+		if _, err := zf.Write(f.data); err != nil {
+			return errors.Wrapf(err, "failed to write to %s: %w", f.name, err)
+		}
+	}
+	return nil
+}
+
+// TODO: Currently external services and site configs are pulled using the SRC_ENDPOINT env var,
+// if theres a way to validate that the env var is pointing at the same instance as the docker and kubectl commands,
+// it should be implemented.
+
+// TODO: file issue on the existence of OAuth signKey which needs to be redacted
+
+// getExternalServicesConfig calls src extsvc list with the format flag -f,
+// and then returns an archiveFile to be consumed
+func getExternalServicesConfig(ctx context.Context, baseDir string) *archiveFile {
+	const fmtStr = `{{range .Nodes}}{{.id}} | {{.kind}} | {{.displayName}}{{"\n"}}{{.config}}{{"\n---\n"}}{{end}}`
+	return archiveFileFromCommand(
+		ctx,
+		filepath.Join(baseDir, "config", "external_services.txt"),
+		os.Args[0], "extsvc", "list", "-f", fmtStr,
+	)
+}
+
+// getSiteConfig calls src api -query=... to query the api for site config json
+func getSiteConfig(ctx context.Context, baseDir string) *archiveFile {
+	const siteConfigStr = `query { site { configuration { effectiveContents } } }`
+	f := archiveFileFromCommand(ctx,
+		filepath.Join(baseDir, "config", "siteConfig.json"),
+		os.Args[0], "api", "-query", siteConfigStr,
+	)
+
+	if f.err != nil {
+		return f
+	}
+
+	var siteConfig struct {
+		Data struct {
+			Site struct {
+				Configuration struct {
+					EffectiveContents string
+				}
+			}
+		}
+	}
+
+	f.err = json.Unmarshal(f.data, &siteConfig)
+	if f.err != nil {
+		return f
+	}
```

Comment:
```suggestion
	if err := json.Unmarshal(f.data, &siteConfig); err != nil {
		f.err = err
		return f
	}
```

nit: I think this reads in a way that keeps intent a little bit more clear.